### PR TITLE
Generate shoot URL in KEB

### PR DESCRIPTION
**Pull request by jasiu001**
Changes proposed in this pull request:

- Shot name cluster and dashboard URL do cluster is generated inside KEB (during trigger provisioning) and sending to provisioner as parameter

### CODE REVIEW

File: components/kyma-environment-broker/internal/broker/instance_create.go [hunk 1]
```diff
@@ -103,12 +118,17 @@ func (b *ProvisionEndpoint) Provision(ctx context.Context, instanceID string, de
...
+	shootName := gardener.CreateShootName()
+	dashboardURL := fmt.Sprintf("https://console.%s.%s.%s", shootName, b.shootProject, strings.Trim(b.shootDomain, "."))
+
 	// create and save new operation
 	operation, err := internal.NewProvisioningOperationWithID(operationID, instanceID, provisioningParameters)
 	if err != nil {
 		logger.Errorf("cannot create new operation: %s", err)
 		return domain.ProvisionedServiceSpec{}, errors.New("cannot create new operation")
 	}
+	operation.ShootName = shootName
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/broker/instance_create.go:131]: in the XSUAA step I need also full domain: "{shootName}.{shootProject}.{shootDomain}"

File: components/kyma-environment-broker/internal/process/provisioning/initialisation.go [hunk 2]
```diff
@@ -209,11 +203,9 @@ func (s *InitialisationStep) handleDashboardURL(instance *internal.Instance, log
...
 	}
 
-	instance.DashboardURL = dashboardURL
-	err = s.instanceStorage.Update(*instance)
-	if err != nil {
-		log.Errorf("cannot update instance: %s", err)
-		return 10 * time.Second, nil
+	if instance.DashboardURL != dashboardURL {
+		log.Errorf("dashboard URL from instance %s is not equal to dashboard URL from director %s", instance.DashboardURL, dashboardURL)
+		return 10 * time.Second, errors.New("dashboard URL from director is incorrect")
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/initialisation.go:208]: if the dashboardURL check fails we need to fail the provisioning operation

**Comment by piotrmiskiewicz** [comment]
prepare also a PR to management-plane-config

File: resources/kcp/charts/kyma-environment-broker/templates/deployment.yaml [hunk 3]
```diff
@@ -273,6 +273,8 @@ spec:
               value: /config/trialRegionMapping.yaml
             - name: APP_GARDENER_PROJECT
               value: {{ .Values.gardener.project }}
+            - name: APP_GARDENER_SHOOT_DOMAIN
```
Comment (ksputo) [inline resources/kcp/charts/kyma-environment-broker/templates/deployment.yaml:276]: Please add this env also to `README.md`

**Review by piotrmiskiewicz — APPROVED** [review]

File: components/kyma-environment-broker/README.md [hunk 4]
```diff
@@ -52,6 +52,9 @@ KEB binary allows you to override some configuration parameters. You can specify
 | **APP_PROVISIONING_MACHINE_IMAGE** | Defines the Gardener machine image used in a provisioned node. | None |
 | **APP_PROVISIONING_MACHINE_IMAGE_VERSION** | Defines the Gardener image version used in a provisioned cluster. | None |
 | **APP_TRIAL_REGION_MAPPING_FILE_PATH** | Defines a path to the file which contains a mapping between the platform region and the Trial plan region. | None |
+| **APP_GARDENER_PROJECT** | Defines the on which the cluster will be created | `kyma-dev` |
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/README.md:55]: maybe: Defines the project on which.....

**Review by ksputo — DISMISSED** [review]

File: components/kyma-environment-broker/README.md [hunk 5]
```diff
@@ -52,6 +52,9 @@ KEB binary allows you to override some configuration parameters. You can specify
 | **APP_PROVISIONING_MACHINE_IMAGE** | Defines the Gardener machine image used in a provisioned node. | None |
 | **APP_PROVISIONING_MACHINE_IMAGE_VERSION** | Defines the Gardener image version used in a provisioned cluster. | None |
 | **APP_TRIAL_REGION_MAPPING_FILE_PATH** | Defines a path to the file which contains a mapping between the platform region and the Trial plan region. | None |
+| **APP_GARDENER_PROJECT** | Defines the project on which the cluster will be created | `kyma-dev` |
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md:55]:
```suggestion
| **APP_GARDENER_PROJECT** | Defines the project in which the cluster is created. | `kyma-dev` |
```

File: components/kyma-environment-broker/README.md [hunk 6]
```diff
@@ -52,6 +52,9 @@ KEB binary allows you to override some configuration parameters. You can specify
...
+| **APP_GARDENER_SHOOT_DOMAIN** | Defines the domain for clusters created on Gardener | `shoot.canary.k8s-hana.ondemand.com` |
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md:56]:
```suggestion
| **APP_GARDENER_SHOOT_DOMAIN** | Defines the domain for clusters created in Gardener. | `shoot.canary.k8s-hana.ondemand.com` |
```

File: components/kyma-environment-broker/README.md [hunk 7]
```diff
@@ -52,6 +52,9 @@ KEB binary allows you to override some configuration parameters. You can specify
...
+| **APP_GARDENER_KUBECONFIG_PATH** | Defines the path to kubeconfig for Gardener | `/gardener/kubeconfig/kubeconfig` |
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md:57]:
```suggestion
| **APP_GARDENER_KUBECONFIG_PATH** | Defines the path to the kubeconfig file for Gardener. | `/gardener/kubeconfig/kubeconfig` |
```

**Review by klaudiagrz — APPROVED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**